enumerables.rb

```diff
@@ -0,0 +1,67 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    length.times { |i| yield(self[i]) }
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    length.times { |i| yield(self[i], i) }
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    array = []
+    my_each { |i| array << i if yield(i) }
+    array
+  end
+
+  def my_all?
+    output = true
+    my_each { |i| break output = false unless yield(i) }
+    output
+  end
+
+  def my_any?
+    output = false
+    my_each { |i| break output = true unless yield(i) }
+    output
+  end
+
+  def my_none?
+    output = true
+    my_each { |i| break output = false if yield(i) }
+    output
+  end
+
+  def my_count(arg)
+    count = 0
+    my_each { |i| count += 1 if arg == i }
+    count
+  end
+
+  def my_map(proc = nil)
+    mapped_arr = []
+    block_given?
+    my_each { |i| mapped_arr << yield(i) }
+    my_each { |i| mapped_arr proc.call(i) }
+    mapped_arr
+  end
```

Comment:
- returns a new array with the results of running a given block once for every element in enum
- returns a new array with the results of running a given block once for every element in enum
- returns an Enumerator if no block is given
- accepts a proc as an argument and returns a new array with the results of running the proc once for every element in enum.